build.sh

```diff
@@ -79,14 +92,28 @@ if [[ ${MAKE_CLEAN} == "ON" ]]; then
   exit 0
 fi
 
+if [[ "${MACHINE}" == "Mac"  ]]; then
+    CMAKE_CMD="cmake -DBUILD_UNIT_TEST=${BUILD_UNITTEST} \
+    -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}
+    -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
+    -DCMAKE_CUDA_COMPILER=${CUDA_COMPILER} \
+    -DMILVUS_ENABLE_PROFILING=${SUPPORT_PROFILING} \
+    -DKNOWHERE_GPU_VERSION=${SUPPORT_GPU} \
+    -DCMAKE_CROSSCOMPILING=true \
```

Comment:
These variables should be set in CMakeLists.txt:
CMAKE_CROSSCOMPILING
RUN_HAVE_GNU_POSIX_REGEX
CMAKE_C_COMPILER
CMAKE_CXX_COMPILER

Reply:
Done.